### PR TITLE
ARROW-16846: [Website] June 2022 Rust Apache Arrow and Parquet 16.0.0 Highlights

### PR DESCRIPTION
**Pull request by alamb**
Closes https://github.com/apache/arrow-rs/issues/1808

### CODE REVIEW

File: _posts/2022-06-11-rust-16.0.0.md [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+
+Some Major Highlights:
+
+* **Advanced Metadata Access**: APIs have been added to access more advanced metadata, such as [PageEncoding](https://github.com/apache/arrow-rs/pull/1322), [BloomFilters](https://github.com/apache/arrow-rs/pull/1309) and [PageIndex](https://github.com/apache/arrow-rs/pull/1762).
+* **Improved usability API**: For example, the [parquet writer now uses `std:io::Write`](https://github.com/apache/arrow-rs/pull/1719)) rather custom `ParquetWriter` trait , making it more interoperable with the rest of the Rust ecosystem and the [projection API is easier to use with nested types](https://github.com/apache/arrow-rs/pull/1716).
+* **Rewritten support for nested types (e.g. struct, lists)** : @tustvold has revamped / rewritten support for [reading](https://github.com/apache/arrow-rs/pull/1682_ and [writing](https://github.com/apache/arrow-rs/pull/1746) structured types, improving the support for arbitrary nested schemas, and is 30% faster.
+
+(TODO maybe a picture / example code to make this blog post less textful)
```
Comment (tustvold) [inline _posts/2022-06-11-rust-16.0.0.md:55]: Just highlighting this is still here 😅

File: _posts/2022-06-11-rust-16.0.0.md [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+
+# Arrow
+
+The Rust arrow implementation has also had substantial improvements, in addition to bug fixes and performance improvements.
+
+Some Major Highlights:
+
+* **Ecosystem Compatibility**: @viriya has put in a massive effort to improve (and prove) compatibility with other Arrow implementations via the rust IPC integration tests. There have been major improvements for corner cases involving nested structures, nullabulity, nested dictionaries, etc.
+* **Safety**: We continue to improve the safety of arrow (checkout README https://github.com/apache/arrow-rs/tree/master/arrow#safety and the module level rustdocs TODO linkaftr 16.0.0 is released ). Among other things, we have added additional validation checking to string kernels https://github.com/apache/arrow-rs/issues/1575 and DecimalArrays (https://github.com/apache/arrow-rs/pull/1767).
+* **Performance**: There have been several major performance improvements such as much caster filter kernels https://github.com/apache/arrow-rs/issues/1288- thanks  @tustvold)
```
Comment (tustvold) [inline _posts/2022-06-11-rust-16.0.0.md:71]:
```suggestion
* **Performance**: There have been several major performance improvements such as much faster filter kernels https://github.com/apache/arrow-rs/issues/1288- thanks  @tustvold)
```

File: _posts/2022-06-11-rust-16.0.0.md [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+* **Parallel Decode**: We intend to transparently support high performance parallel decoding of parquet to arrow arrays, when invoked from a [rayon](https://crates.io/crates/rayon) threadpool.
+
+# Arrow
+
+The Rust arrow implementation has also had substantial improvements, in addition to bug fixes and performance improvements.
+
+Some Major Highlights:
+
+* **Ecosystem Compatibility**: @viriya has put in a massive effort to improve (and prove) compatibility with other Arrow implementations via the rust IPC integration tests. There have been major improvements for corner cases involving nested structures, nullabulity, nested dictionaries, etc.
+* **Safety**: We continue to improve the safety of arrow (checkout README https://github.com/apache/arrow-rs/tree/master/arrow#safety and the module level rustdocs TODO linkaftr 16.0.0 is released ). Among other things, we have added additional validation checking to string kernels https://github.com/apache/arrow-rs/issues/1575 and DecimalArrays (https://github.com/apache/arrow-rs/pull/1767).
```
Comment (tustvold) [inline _posts/2022-06-11-rust-16.0.0.md:70]: Another todo

**Comment by alamb** [comment]
> @alamb I couldn't push my changes into your branch, so I've opened a PR instead: https://github.com/alamb/arrow-site/pull/2

Thanks, and sorry @nevi-me  -- I didn't do that on purpose 🤔
